Marlin/src/feature/pause.cpp

```diff
@@ -298,48 +331,104 @@ void resume_print(const float &load_length/*=0*/, const float &initial_extrude_l
-
-      extrude_length = ADVANCED_PAUSE_EXTRUDE_LENGTH;
-
-      // Keep looping if "Extrude More" was selected
-    } while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_EXTRUDE_MORE);
-
+    if (swap_spool_disabled){ 
+        
+	  do {
+	    if (extrude_length > 0) {
+		  // "Wait for filament extrude"
+		  if (swap_spool_disabled) lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_EXTRUDE); 
+		  // Extrude filament to get into hotend
+          do_pause_e_move(extrude_length, ADVANCED_PAUSE_EXTRUDE_FEEDRATE);
+	    }
+		// Show "Extrude More" / "Resume" menu and wait for reply
+		KEEPALIVE_STATE(PAUSED_FOR_USER);
+		wait_for_user = false;
+		lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_OPTION);
+	    while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_WAIT_FOR) idle(true);
+		KEEPALIVE_STATE(IN_HANDLER);
+		
+		extrude_length = ADVANCED_PAUSE_EXTRUDE_LENGTH;
+		// Keep looping if "Extrude More" was selected
+	  } while (advanced_pause_menu_response == ADVANCED_PAUSE_RESPONSE_EXTRUDE_MORE);
+  }
+  else do_pause_e_move(extrude_length, ADVANCED_PAUSE_EXTRUDE_FEEDRATE);
   #endif
-
+  
+  // Extruder purged = initialisation  
+  planner.set_e_position_mm(0); current_position[E_AXIS] = 0; destination[E_AXIS]=0;
+  
   #if ENABLED(ULTIPANEL)
     // "Wait for print to resume"
-    lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_RESUME);
+    if (swap_spool_disabled)lcd_advanced_pause_show_message(ADVANCED_PAUSE_MESSAGE_RESUME);
   #endif
-
-  // Set extruder to saved position
-  planner.set_e_position_mm((current_position[E_AXIS] = resume_position[E_AXIS]));
-
+	
+  // Stop blowing after delay
+  #ifdef ADVANCED_PAUSE_DWELL
+    delay(ADVANCED_PAUSE_DWELL);
+  #endif
+  #if (ADVANCED_PAUSE_FANSPEED >0 && ADVANCED_PAUSE_FAN < FAN_COUNT)
+    fanSpeeds[ADVANCED_PAUSE_FAN]=0 ;
+  #endif  
+ 
+  // Intelligent Retractation before return to print
+  // If resume position negative then travel back Retraction = resume_position
+  // Rare but possible : can go on pause with retracted/swapped status or negative position
+  #ifdef PAUSE_PARK_RETRACT_LENGTH
+    #if ENABLED(FWRETRACT)
+	  //If retracted/swapped goto negative position
+	  if (fwretract.retracted_swap[active_extruder]) {
+		const float tmp_length=-fwretract.swap_retract_length;
+	    const float tmp_feed=fwretract.retract_feedrate_mm_s;
+	    do_pause_e_move(tmp_length, tmp_feed); 
+	  }
+	  // If retracted or not : normal fwretract used
+	  else {
+	   const float tmp_length=-fwretract.retract_length;  
+       const float tmp_feed=fwretract.retract_feedrate_mm_s;   
+       do_pause_e_move(tmp_length, tmp_feed); 
+	 }
+    #else
+	  //If negative and more than PAUSE_PARK_RETRACT_LENGTH goto resume_position 
+	  if (resume_position[E_AXIS]<(-PAUSE_PARK_RETRACT_LENGTH)) do_pause_e_move(resume_position[E_AXIS], PAUSE_PARK_RETRACT_FEEDRATE);
+      else do_pause_e_move(-PAUSE_PARK_RETRACT_LENGTH, PAUSE_PARK_RETRACT_FEEDRATE);
+    #endif
+  #endif	
+ 
```

Comment:
INTELLIGENT RESUMING
Before return to print , we retract , but we retract with intelligence
If old resume ' e ' position is negative or fwretracted, we use this position  ' and we will not need to change it when return to print'
If old resume ' e ' position if positive , we retract with the PAUSE_PARK_RETRACT_LENGTH defined value or the current FW retract value

VERY GOOD !!!